specification/network/resource-manager/Microsoft.Network/stable/2024-07-01/networkSecurityPerimeter.json

```diff
@@ -0,0 +1,2386 @@
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Request is successful. Returns a list of network security perimeters.",
+            "schema": {
+              "$ref": "#/definitions/NetworkSecurityPerimeterListResult"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "./network.json#/definitions/CloudError"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "List Network Security Perimeter": {
+            "$ref": "./examples/NetworkSecurityPerimeterList.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Network/networkSecurityPerimeters/{networkSecurityPerimeterName}/profiles/{profileName}": {
+      "parameters": [
+        {
+          "$ref": "./network.json#/parameters/SubscriptionIdParameter"
+        },
+        {
+          "$ref": "#/parameters/ResourceGroupNameParameter"
+        },
+        {
+          "$ref": "#/parameters/NetworkSecurityPerimeterNameParameter"
+        },
+        {
+          "$ref": "#/parameters/ProfileNameParameter"
+        },
+        {
+          "$ref": "./network.json#/parameters/ApiVersionParameter"
+        }
+      ],
+      "get": {
+        "tags": [
+          "NspProfiles"
+        ],
+        "operationId": "NspProfiles_Get",
+        "description": "Gets the specified NSP profile.",
+        "responses": {
+          "200": {
+            "description": "OK - Returns information about the NSP profile.",
+            "schema": {
+              "$ref": "#/definitions/NspProfile"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "./network.json#/definitions/CloudError"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "NspProfilesGet": {
+            "$ref": "./examples/NspProfileGet.json"
+          }
+        }
+      },
+      "put": {
+        "tags": [
+          "NspProfiles"
+        ],
+        "operationId": "NspProfiles_CreateOrUpdate",
+        "description": "Creates or updates a network profile.",
+        "parameters": [
+          {
+            "name": "parameters",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/NspProfile"
+            },
+            "description": "Parameters that hold the NspProfile resource to be created/updated."
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Update - Profile is immutable resource. Returns the existing profile resource.",
+            "schema": {
+              "$ref": "#/definitions/NspProfile"
+            }
+          },
+          "201": {
```

Comment:
>"201" [](http://example.com/codeflow?start=10&length=5)

since this is async operation add below options to this API

"x-ms-long-running-operation": true,
"x-ms-long-running-operation-options": {
    "final-state-via": "azure-async-operation"
}